PLUGINS.md

```diff
@@ -0,0 +1,180 @@
+# OpenShift Console Plugins
+
+Openshift Console Plugins extend the functionality of the console. Plugins are enabled by
+adding them as `dependencies` of the [console-app](frontend/packages/console-app) package.
+
+See the [demo plugin](frontend/packages/console-demo-plugin) for reference.
+
+## Directory and File Names
+
+Plugins packages should be in [frontend/packages](frontend/packages) directory. A plugin should have
+a `package.json`, `OWNERS`, and a plugin entry file, the plugin entry file is usually
+called `src/plugin.tsx`.
+
+Optionally unit tests will be in `__tests__` directories close to the source code, and 
+integration tests should be in an `integration-tests` directory.
+
+``` bash
+    .
+    ├── ...
+    ├── frontend
+    │   ├── packages
+    |   |   ├── ...
+    │   |   ├── my-plugin
+    |   |   |   ├── integration-tests
+    |   |   |   |   └── ...
+    |   |   |   ├── src
+    |   |   |   |   ├── plugin.tsx
+    |   |   |   |   ├── __tests__
+    |   |   |   |   |   ├── some-test.spec.ts
+    |   |   |   |   |   └── ...
+    |   |   |   |   └── ...
+    |   |   |   ├── OWNERS
+    |   |   |   ├── package.json
+    |   |   |   └── ...
+    |   |   └── ...
+    │   └── ...
+    └── ...
+```
+
+## Plugin Directory
+
+Plugin directory name should match the plugin name (console plugins are an exception), as a convention plugin 
+directory names should end with the `-plugin` suffix (eg. `my-plugin`).
+
+- Use lowercase dash-separated names for all files (to avoid git issues with case-insensitive file systems)
+- Exceptions are files which have their own naming conventions (eg Makefile, README)
+
+## Package.json
+
+Plugins are packages, the package definitions of a plugin should include a `consolePlugin`
```

Comment:
The `consolePlugin.entry` path should point to a module that exports the `Plugin` object (a.k.a plugin entry module).

From plugin author perspective, a `Plugin` is simply a list of extensions.

The plugin entry module is loaded right upon Console startup, so it should lazy-load other code (use the `import` function) instead of directly referring to it.